packages/core-dialog/core-dialog.js

```diff
@@ -25,23 +25,37 @@ export default function dialog (dialogs, open) {
     dialog.setAttribute('aria-modal', true)
     dialog.setAttribute('aria-label', options.label || dialog.getAttribute('aria-label'))
     hasBackdrop || dialog.insertAdjacentElement('afterend', document.createElement('backdrop'))
+    options.strict && dialog.setAttribute('data-dialog-strict', options.strict)
 
     setOpen(dialog, options.open)
     return dialog
   })
 }
 
+// Focus can move outside the dialog when in "strict"-mode, therefore
+// we need to ensure that if that happens we will refocus on the dialog.
+addEvent(UUID, 'focus', (event) => {
+  let hasDialog = false
+  for (let el = event.target; el; el = el.parentElement) {
+    if (el.hasAttribute(UUID)) return (hasDialog = true)
```

Comment:
why do you need the `hasDialog` - as you do `return`, you exit the whole function all together anyway :)